Mention end-inclusiveness in intrinsic F# slices (#16386)
dotnet/docs: docs/fsharp/language-reference/slices.md

---
title: Slices

> description: Learn about how to use slices for existing F# data types and how to define your own slices for other data types.

Reviewer: ```suggestion
description: Learn how to use slices for existing F# data types and how to define your own slices for other data types.
```

ms.date: 12/23/2019
---
# Slices

> In F#, a slice is a subset of any data type that defines a `GetSlice` method in its definition or in a [type extension](type-extensions.md) that is in scope. It is most commonly used with F# arrays and lists. This article explains how to take slices from existing F# types and how to define your own.

Reviewer: ```suggestion
In F#, a slice is a subset of any data type that has a `GetSlice` method in its definition or in an in-scope [type extension](type-extensions.md). It is most commonly used with F# arrays and lists. This article explains how to take slices from existing F# types and how to define your own slices.
```
Wasn't exactly sure what "your own" referred to, so please correct if wrong.

printfn "%A" twoByTwo
```

> The F# core library does not currently define `GetSlice`for 3D arrays. If you wish to slice those or other arrays of more dimensions, you must define the `GetSlice` member yourself.

Reviewer: ```suggestion
The F# core library does not currently define `GetSlice` for 3D arrays. If you wish to slice 3D arrays or other arrays of more dimensions, define the `GetSlice` member yourself.
```

printSpan sp.[1..2] // |2; 3|]
```

## Built-in F# slices are end-inclusive

> All intrinsic slices in F# are end-inclusive; that is, the upper bound is included in the slice. For a given slice with starting index `x` and ending index `y`, the resulting slice will include the `yth` value.

Reviewer: ```suggestion
All intrinsic slices in F# are end-inclusive; that is, the upper bound is included in the slice. For a given slice with starting index `x` and ending index `y`, the resulting slice will include the *yth* value.
```
"th" seems English-specific, so it should probably be localized (so I took it out of backticks).
